packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -233,30 +234,63 @@ export class MultistepDialog extends AbstractPureComponent2<MultistepDialogProps
         if (this.state.selectedIndex > 0) {
             const backButtonProps = steps[selectedIndex].props.backButtonProps ?? this.props.backButtonProps;
 
-            buttons.push(
+            if (backButtonProps?.tooltipContent) {
+                buttons.push(
+                    <Tooltip2 content={backButtonProps?.tooltipContent} hoverOpenDelay={100}>
+                        <Button
+                            key="back"
+                            onClick={this.getDialogStepChangeHandler(selectedIndex - 1)}
+                            text="Back"
+                            {...backButtonProps}
+                        />
+                    </Tooltip2>,
+                );
+            } else {
                 <Button
                     key="back"
                     onClick={this.getDialogStepChangeHandler(selectedIndex - 1)}
                     text="Back"
                     {...backButtonProps}
-                />,
-            );
+                />;
+            }
         }
 
         if (selectedIndex === this.getDialogStepChildren().length - 1) {
-            buttons.push(<Button intent="primary" key="final" text="Submit" {...this.props.finalButtonProps} />);
+            if (this.props.finalButtonProps?.tooltipContent) {
```

Comment:
same here, please avoid implicit type coercion, and use an explicit check instead:

```suggestion
            if (this.props.finalButtonProps?.tooltipContent != null) {
```